### ISSUE TITLE
Building a system with martini Protein/lipid and new coarse grained water model.

### ISSUE THREAD

**Issue by pearlDingzhen** [maccallumlab/martini_openmm#11]
Hello!
My supervisor developed a coarse-grained water model, namely flexDQ(doi: 10.1039/D0CP04185H), which is a flexible three-site coarse-grained model represent 4 water molecules and coupled it with martini protein and lipid model. The interaction between water molecules are defined by tabulated potential and interaction between water molecules and martini beads are just the same way as Martini one-site moleculer do: the 'Oxygen' with negative charge interact with martini beads by a LJ potential defined by C6 and C12 parameter while the 'Hydrogen' is invisible to the martini protein and lipids.
We developed this model on Gromacs. However, the gromacs can not support tabulated potential run on GPUs. So I want to run it with OpenMM. I have convert the topology and tabulated potential of my water model from gromacs file to openMM xml format forcefield file. And I can build a pure water simulation object from pdb and xml file by forcefield createsystem method. 
Now I want to run Martini-flexdq system. So i use the martini_openmm to read Martini gromacs file. Your MartiniTopFile object create_system method directly return a system object. So I have two system object: one is created by forcefield createsystem method, another is created by your MartiniTopFile.create_system. How can I combine the two system together with proper nonbonded interaction parameter between my water model and martini protein/lipids?

**Comment by jlmaccal** [comment]
I don't think you can combine these easily. OpenMM doesn't really have any built-in facilities for combining systems or mixing force fields, as this isn't a very well-defined operation. I think it would be very error-prone for you to try to implement this.

I think the most straightforward approach would be to implement the flexible water model as a GROMACS topology that can be directly read by the Martini parser in OpenMM.

**Comment by pearlDingzhen** [comment]
I forgot to check my email and very grateful for your kind reply.

发件人：Justin MacCallum ***@***.***>
发送日期：2023-03-03 00:26:46
收件人：maccallumlab/martini_openmm ***@***.***>
抄送人：pearlDingzhen ***@***.***>,Author ***@***.***>
主题：Re: [maccallumlab/martini_openmm] Building a system with martini Protein/lipid and new coarse grained water model. (Issue #11)

I don't think you can combine these easily. OpenMM doesn't really have any built-in facilities for combining systems or mixing force fields, as this isn't a very well-defined operation. I think it would be very error-prone for you to try to implement this.
I think the most straightforward approach would be to implement the flexible water model as a GROMACS topology that can be directly read by the Martini parser in OpenMM.
